modules/storages/lib/api/v3/file_links/create_endpoint.rb

```diff
@@ -0,0 +1,38 @@
+module API::V3::FileLinks
+  class CreateEndpoint < API::Utilities::Endpoints::Create
+    include ::API::V3::Utilities::Endpoints::V3Deductions
+    include ::API::V3::Utilities::Endpoints::V3PresentSingle
+
+    def process(request, params_elements)
+      global_result = ServiceResult.new(
+        success: true,
+        result: []
+      )
+      params_elements.each do |params|
+        one_result = super(request, params)
+        global_result.add_dependent!(one_result)
```

Comment:
`add_dependent!` will merge two `ServiceResult` instances which is useful one Service is orchestrating a bunch of other Services, like this one does here.

Since `add_dependent!` will also add the `result` of the passed to `ServiceResult` to the array of results, the next line is superfluous.

Reply:
In fact `add_dependent!` will add the `dependent` object to its own `dependent_results` array, which is not the same. I saw nothing that fetches all results of all dependent service calls into one single array and maybe I overlooked it. I was not totally sure about what `add_dependent!` was doing.

Reply:
@cbliard [#all_results](https://github.com/opf/openproject/blob/dev/app/services/service_result.rb#L85-L89) should work if I am not mistaken.

Reply:
Indeed. Done in 2ec5db6e89.